src/NSwag.SwaggerGeneration.AspNetCore/AspNetCoreToSwaggerGenerator.cs

```diff
@@ -0,0 +1,257 @@
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Mvc.ApiExplorer;
+using Microsoft.AspNetCore.Mvc.Controllers;
+using NJsonSchema;
+using NJsonSchema.Infrastructure;
+using NSwag.SwaggerGeneration.Processors.Contexts;
+
+namespace NSwag.SwaggerGeneration.AspNetCore
+{
+    /// <summary>Generates a <see cref="SwaggerDocument"/> object for the given Web API class type. </summary>
+    public class AspNetCoreToSwaggerGenerator
+    {
+        private readonly SwaggerJsonSchemaGenerator _schemaGenerator;
+
+        /// <summary>Initializes a new instance of the <see cref="AspNetCoreToSwaggerGenerator" /> class.</summary>
+        /// <param name="settings">The settings.</param>
+        public AspNetCoreToSwaggerGenerator(AspNetCoreToSwaggerGeneratorSettings settings)
+            : this(settings, new SwaggerJsonSchemaGenerator(settings))
+        {
+        }
+
+        /// <summary>Initializes a new instance of the <see cref="AspNetCoreToSwaggerGenerator" /> class.</summary>
+        /// <param name="settings">The settings.</param>
+        /// <param name="schemaGenerator">The schema generator.</param>
+        public AspNetCoreToSwaggerGenerator(AspNetCoreToSwaggerGeneratorSettings settings, SwaggerJsonSchemaGenerator schemaGenerator)
+        {
+            Settings = settings;
+            _schemaGenerator = schemaGenerator;
+        }
+
+        /// <summary>Gets or sets the generator settings.</summary>
+        public AspNetCoreToSwaggerGeneratorSettings Settings { get; }
+
+        /// <summary>Generates a Swagger specification for the given <see cref="ApiDescriptionGroupCollection"/>.</summary>
+        /// <param name="apiDescriptionGroups"><see cref="ApiDescriptionGroup"/>.</param>
+        /// <returns>The <see cref="SwaggerDocument" />.</returns>
+        /// <exception cref="InvalidOperationException">The operation has more than one body parameter.</exception>
+        public Task<SwaggerDocument> GenerateAsync(ApiDescriptionGroupCollection apiDescriptionGroups)
+        {
+            var apiDescriptions = apiDescriptionGroups.Items.SelectMany(g => g.Items);
+            return GenerateAsync(apiDescriptions);
+        }
+
+        /// <summary>
+        /// Generates a Swagger specification for the given <see cref="ApiDescription"/> sequence.
+        /// </summary>
+        /// <param name="apiDescriptions"><see cref="ApiDescription"/></param>
+        /// <returns>The <see cref="SwaggerDocument" />.</returns>
+        public async Task<SwaggerDocument> GenerateAsync(IEnumerable<ApiDescription> apiDescriptions)
+        {
+            var document = await CreateDocumentAsync(Settings).ConfigureAwait(false);
+            var schemaResolver = new SwaggerSchemaResolver(document, Settings);
+
+            var apiGroups = apiDescriptions.Where(apiDescription => apiDescription.ActionDescriptor is ControllerActionDescriptor)
```

Comment:
use `OfType`

Reply:
It's a subproperty and we want to get the top level thing. Btw, this was an open question. Should we attempt to generate documents for ApiDescriptions with the available information if we aren't dealing with `ControllerActionDescriptor` instances?

Reply:
Ah gotcha.  Ignore me then.

> Should we attempt to generate documents for ApiDescriptions with the available information if we aren't dealing with ControllerActionDescriptor instances?

NSwag requires a method and class - therefore `ControllerActionDescriptor` is required.